tests/test_routes.py

```diff
@@ -91,3 +94,117 @@ def test_index(self):
         """ It should call the home page """
         resp = self.client.get("/")
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_get_shopcart(self):
+        """It should Read a single Shopcart"""
+        # get the id of a shopcart
+        test_shopcart = self._create_an_empty_shopcart(1)[0]
+        resp = self.client.get(
+            f"{BASE_URL}/{test_shopcart.id}", content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+        self.assertEqual(data["id"], test_shopcart.id)
+
+    def test_get_shopcart_not_found(self):
+        """It should not Read a Shopcart that is not found"""
+        response = self.client.get(f"{BASE_URL}/0")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+    
+    # @patch.object(Shopcart, 'get_by_id', MagicMock(side_effect=Exception("DBAPIErr")))
```

Comment:
could you help remove the commented code lines not in used below?

Reply:
done